### PR TITLE
task: Fix validations for restore

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

While slowly enabling https://github.com/Djarvur/go-err113 I spotted a couple of artefacts for restore commands 

- cluster ID is not part of atlas only cloud manager/ops manager
- cluster name can now always be mandatory
- drive by fixing examples

I manually tested the examples work

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/backup/backup.go [hunk 1]
```diff
@@ -35,3 +35,19 @@ func Builder() *cobra.Command {
 
 	return cmd
 }
+
+func RestoresBuilder() *cobra.Command {
```
Comment (gssbzn) [inline internal/cli/atlas/backup/backup.go:39]: moved from a different file

File: internal/cli/atlas/backup/restores_start.go [hunk 2]
```diff
@@ -139,11 +130,7 @@ func markRequiredPointInTimeRestoreFlags(cmd *cobra.Command) error {
 		return err
 	}
 
-	if err := cmd.MarkFlagRequired(flag.TargetClusterID); err != nil {
```
Comment (gssbzn) [inline internal/cli/atlas/backup/restores_start.go:142]: cluster id is a cloud manager/ops manager thing

File: docs/atlascli/command/atlas-backups-restores-list.txt [hunk 3]
```diff
@@ -85,3 +85,11 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example etrieve the Continuous Backup restore jobs for the cluster Cluster0:
```
Comment (MihaiBojin) [inline docs/atlascli/command/atlas-backups-restores-list.txt:93]: `retrieves` not etrieve.

Also, do we somehow mark names, e.g. **Cluster0** or *Cluster0*?

File: docs/mongocli/command/mongocli-atlas-backups-restores-list.txt [hunk 4]
```diff
@@ -85,3 +85,11 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example etrieve the Continuous Backup restore jobs for the cluster Cluster0:
```
Comment (MihaiBojin) [inline docs/mongocli/command/mongocli-atlas-backups-restores-list.txt:93]: same comment, misspelled retrieves

File: internal/cli/atlas/backup/restores_list.go [hunk 5]
```diff
@@ -66,8 +67,11 @@ func RestoresListBuilder() *cobra.Command {
 		Args:    require.ExactArgs(1),
 		Annotations: map[string]string{
 			"args":            "clusterName",
+			"requiredArgs":    "clusterName",
 			"clusterNameDesc": "Name of the Atlas cluster for which you want to retrieve restore jobs.",
 		},
+		Example: fmt.Sprintf(`The following example etrieve the Continuous Backup restore jobs for the cluster Cluster0:
```
Comment (MihaiBojin) [inline internal/cli/atlas/backup/restores_list.go:73]: same err

File: internal/cli/atlas/backup/restores_start.go (hunk 2 above)
Comment (MihaiBojin) [reply internal/cli/atlas/backup/restores_start.go:142]: Is this not the same command that is used for `mongocli`?

File: internal/cli/global_opts.go [hunk 6]
```diff
@@ -105,3 +105,18 @@ func GenerateAliases(use string, extra ...string) []string {
 	aliases = append(aliases, extra...)
 	return aliases
 }
+
+var exampleBin string
+
+// ExampleAtlasEntryPoint returns the entry point for an atlas command while taking into account
+// if the bin is atlas or mongocli.
+func ExampleAtlasEntryPoint() string {
```
Comment (MihaiBojin) [inline internal/cli/global_opts.go:113]: nice!

File: internal/cli/atlas/backup/restores_start.go (hunk 2 above)
Comment (gssbzn) [reply internal/cli/atlas/backup/restores_start.go:142]: atlas specific, at some point it was reused for the 3 (or a version of it) and where I believe the mix came from

File: docs/atlascli/command/atlas-backups-restores-list.txt (hunk 3 above)
Comment (gssbzn) [reply docs/atlascli/command/atlas-backups-restores-list.txt:93]: I'll let @mongodb/docs-cloud-team decide on formatting, I think I copy pasted the text from an old mongocli docs page

File: docs/atlascli/command/atlas-backups-restores-list.txt [hunk 7]
```diff
@@ -85,3 +85,11 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example retrieve the Continuous Backup restore jobs for the cluster Cluster0:
```
Comment (sarahsimpers) [inline docs/atlascli/command/atlas-backups-restores-list.txt:93]:
```suggestion
 The following example retrieves the continuous backup restore jobs for the cluster Cluster0:
```

File: docs/atlascli/command/atlas-backups-restores-start.txt [hunk 8]
```diff
@@ -105,3 +105,27 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example creates an automated restore
```
Comment (sarahsimpers) [inline docs/atlascli/command/atlas-backups-restores-start.txt:113]:
```suggestion
 The following example creates an automated restore:
```

File: docs/atlascli/command/atlas-backups-restores-start.txt [hunk 9]
```diff
@@ -105,3 +105,27 @@ Inherited Options
...
+   $ atlas backup restore start automated \
+          --clusterName myDemo \
+          --snapshotId 5e7e00128f8ce03996a47179 \
+          --targetClusterName myDemo2 \
+          --targetProjectId 1a2345b67c8e9a12f3456de7
+
+ The following example creates a point in time restore
```
Comment (sarahsimpers) [inline docs/atlascli/command/atlas-backups-restores-start.txt:120]:
```suggestion
 The following example creates a point-in-time restore:
```

File: docs/atlascli/command/atlas-backups-restores-start.txt [hunk 10]
```diff
@@ -105,3 +105,27 @@ Inherited Options
...
+   $ atlas backup restore start pointInTime \
+          --clusterName myDemo \
+          --pointInTimeUTCMillis 1588523147 \
+          --targetClusterName myDemo2 \
+          --targetProjectId 1a2345b67c8e9a12f3456de7
+   
+ The following example creates a point in time restore
```
Comment (sarahsimpers) [inline docs/atlascli/command/atlas-backups-restores-start.txt:127]: Question: should this be download?
```suggestion
 The following example creates a download restore:
```

File: docs/mongocli/command/mongocli-atlas-backups-restores-list.txt [hunk 11]
```diff
@@ -85,3 +85,11 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example retrieve the Continuous Backup restore jobs for the cluster Cluster0:
```
Comment (sarahsimpers) [inline docs/mongocli/command/mongocli-atlas-backups-restores-list.txt:93]:
```suggestion
 The following example retrieves the continuous backup restore jobs for the cluster Cluster0:
```

File: docs/mongocli/command/mongocli-atlas-backups-restores-start.txt [hunk 12]
```diff
@@ -105,3 +105,27 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example creates an automated restore
```
Comment (sarahsimpers) [inline docs/mongocli/command/mongocli-atlas-backups-restores-start.txt:113]:
```suggestion
 The following example creates an automated restore:
```

File: docs/mongocli/command/mongocli-atlas-backups-restores-start.txt [hunk 13]
```diff
@@ -105,3 +105,27 @@ Inherited Options
...
+   $ mongocli atlas backup restore start automated \
+          --clusterName myDemo \
+          --snapshotId 5e7e00128f8ce03996a47179 \
+          --targetClusterName myDemo2 \
+          --targetProjectId 1a2345b67c8e9a12f3456de7
+
+ The following example creates a point in time restore
```
Comment (sarahsimpers) [inline docs/mongocli/command/mongocli-atlas-backups-restores-start.txt:120]:
```suggestion
 The following example creates a point-in-time restore:
```

File: docs/mongocli/command/mongocli-atlas-backups-restores-start.txt [hunk 14]
```diff
@@ -105,3 +105,27 @@ Inherited Options
...
+   $ mongocli atlas backup restore start pointInTime \
+          --clusterName myDemo \
+          --pointInTimeUTCMillis 1588523147 \
+          --targetClusterName myDemo2 \
+          --targetProjectId 1a2345b67c8e9a12f3456de7
+   
+ The following example creates a point in time restore
```
Comment (sarahsimpers) [inline docs/mongocli/command/mongocli-atlas-backups-restores-start.txt:127]:
```suggestion
 The following example creates a download restore:
```

File: internal/cli/atlas/backup/restores_list.go [hunk 15]
```diff
@@ -66,8 +67,11 @@ func RestoresListBuilder() *cobra.Command {
 		Args:    require.ExactArgs(1),
 		Annotations: map[string]string{
 			"args":            "clusterName",
+			"requiredArgs":    "clusterName",
 			"clusterNameDesc": "Name of the Atlas cluster for which you want to retrieve restore jobs.",
 		},
+		Example: fmt.Sprintf(`The following example retrieve the Continuous Backup restore jobs for the cluster Cluster0:
```
Comment (sarahsimpers) [inline internal/cli/atlas/backup/restores_list.go:73]:
```suggestion
		Example: fmt.Sprintf(`The following example retrieves the Continuous Backup restore jobs for the cluster Cluster0:
```

File: internal/cli/atlas/backup/restores_start.go [hunk 16]
```diff
@@ -156,8 +143,27 @@ func RestoresStartBuilder() *cobra.Command {
 		ValidArgs: []string{automatedRestore, downloadRestore, pointInTimeRestore},
 		Annotations: map[string]string{
 			"args":             "deliveryType",
+			"requiredArgs":     "deliveryType",
 			"deliveryTypeDesc": "Type of restore job to create. Accepted values include: automated, download, pointInTime.",
 		},
+		Example: fmt.Sprintf(`The following example creates an automated restore
+  $ %[1]s backup restore start automated \
```
Comment (sarahsimpers) [inline internal/cli/atlas/backup/restores_start.go:150]: Not sure if this file requires the same changes as above. If so, please make them here too.

**Review by sarahsimpers — CHANGES_REQUESTED** [review]
Just some small changes. Thanks!

File: docs/atlascli/command/atlas-backups-restores-list.txt [hunk 17]
```diff
@@ -85,3 +85,11 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example retrieves the continuous backup restore jobs for the cluster Cluster0::
```
Comment (sarahsimpers) [inline docs/atlascli/command/atlas-backups-restores-list.txt:93]: Oddly, the second ":" was not showing up before so now there's doubles... sorry to make you change twice. Please remove one
```suggestion
 The following example retrieves the continuous backup restore jobs for the cluster Cluster0:
```

File: docs/mongocli/command/mongocli-atlas-backups-restores-list.txt [hunk 18]
```diff
@@ -85,3 +85,11 @@ Inherited Options
      - false
      - Profile to use from your configuration file.
 
+Examples
+--------
+
+.. code-block::
+
+ The following example retrieves the continuous backup restore jobs for the cluster Cluster0::
```
Comment (sarahsimpers) [inline docs/mongocli/command/mongocli-atlas-backups-restores-list.txt:93]:
```suggestion
 The following example retrieves the continuous backup restore jobs for the cluster Cluster0:
```

File: internal/cli/atlas/backup/restores_list.go [hunk 19]
```diff
@@ -66,8 +67,11 @@ func RestoresListBuilder() *cobra.Command {
 		Args:    require.ExactArgs(1),
 		Annotations: map[string]string{
 			"args":            "clusterName",
+			"requiredArgs":    "clusterName",
 			"clusterNameDesc": "Name of the Atlas cluster for which you want to retrieve restore jobs.",
 		},
+		Example: fmt.Sprintf(`The following example retrieves the continuous backup restore jobs for the cluster Cluster0::
```
Comment (sarahsimpers) [inline internal/cli/atlas/backup/restores_list.go:73]:
```suggestion
		Example: fmt.Sprintf(`The following example retrieves the continuous backup restore jobs for the cluster Cluster0:
```

**Review by sarahsimpers — APPROVED** [review]
LGTM % change to remove double colons (not sure if I just missed it the first time or what, sorry!)

File: internal/cli/atlas/backup/restores_list.go [hunk 20]
```diff
@@ -66,8 +67,11 @@ func RestoresListBuilder() *cobra.Command {
 		Args:    require.ExactArgs(1),
 		Annotations: map[string]string{
 			"args":            "clusterName",
+			"requiredArgs":    "clusterName",
 			"clusterNameDesc": "Name of the Atlas cluster for which you want to retrieve restore jobs.",
 		},
+		Example: fmt.Sprintf(`The following example retrieves the continuous backup restore jobs for the cluster Cluster0:
+  $ %s backup restore list Cluster0`, cli.ExampleAtlasEntryPoint()),
```
Comment (andreaangiolillo) [inline internal/cli/atlas/backup/restores_list.go:74]: Could you update the examples in `atlas cluster create`, `atlas cluster update`, `dbuser create`, `accesslist create ` to use `cli.ExampleAtlasEntryPoint()`

Comment (gssbzn) [reply internal/cli/atlas/backup/restores_list.go:74]: > Could you update the examples in atlas cluster create, atlas cluster update, dbuser create, accesslist create to use cli.ExampleAtlasEntryPoint()?

Can we do that as a separate PR?

Comment (andreaangiolillo) [reply internal/cli/atlas/backup/restores_list.go:74]: sure

**Pull request merged**